ext/xmlreader/php_xmlreader.stub.php

```diff
@@ -33,77 +33,77 @@ class XMLReader
+    /** @tentative-return-type */
+    public function isValid(): bool {}
 
-    /** @return string|null */
-    public function lookupNamespace(string $prefix) {}
+    /** @tentative-return-type */
+    public function lookupNamespace(string $prefix): ?string {}
 
-    /** @return bool */
-    public function moveToAttribute(string $name) {}
+    /** @tentative-return-type */
+    public function moveToAttribute(string $name): bool {}
 
-    /** @return bool */
-    public function moveToAttributeNo(int $index) {}
+    /** @tentative-return-type */
+    public function moveToAttributeNo(int $index): bool {}
 
-    /** @return bool */
-    public function moveToAttributeNs(string $name, string $namespace) {}
+    /** @tentative-return-type */
+    public function moveToAttributeNs(string $name, string $namespace): bool {}
 
-    /** @return bool */
-    public function moveToElement() {}
+    /** @tentative-return-type */
+    public function moveToElement(): bool {}
 
-    /** @return bool */
-    public function moveToFirstAttribute() {}
+    /** @tentative-return-type */
+    public function moveToFirstAttribute(): bool {}
 
-    /** @return bool */
-    public function moveToNextAttribute() {}
+    /** @tentative-return-type */
+    public function moveToNextAttribute(): bool {}
 
-    /** @return bool */
-    public function read() {}
+    /** @tentative-return-type */
+    public function read(): bool {}
 
-    /** @return bool */
-    public function next(?string $name = null) {}
+    /** @tentative-return-type */
+    public function next(?string $name = null): bool {}
 
     /** @return bool|XMLReader */
     public static function open(string $uri, ?string $encoding = null, int $flags = 0) {}
 
-    /** @return string */
-    public function readInnerXml() {}
+    /** @tentative-return-type */
+    public function readInnerXml(): string {}
 
-    /** @return string */
-    public function readOuterXml() {}
+    /** @tentative-return-type */
+    public function readOuterXml(): string {}
 
-    /** @return string */
-    public function readString() {}
+    /** @tentative-return-type */
+    public function readString(): string {}
 
-    /** @return bool */
-    public function setSchema(?string $filename) {}
+    /** @tentative-return-type */
+    public function setSchema(?string $filename): bool {}
 
-    /** @return bool */
-    public function setParserProperty(int $property, bool $value) {}
+    /** @tentative-return-type */
+    public function setParserProperty(int $property, bool $value): bool {}
 
-    /** @return bool */
-    public function setRelaxNGSchema(?string $filename) {}
+    /** @tentative-return-type */
+    public function setRelaxNGSchema(?string $filename): bool {}
 
-    /** @return bool */
-    public function setRelaxNGSchemaSource(?string $source) {}
+    /** @tentative-return-type */
+    public function setRelaxNGSchemaSource(?string $source): bool {}
 
     /** @return bool|XMLReader */
```

Comment:
Why were the static methods not changed?

Reply:
Ah, yes, that's because I wasn't sure that the return types makes sense/needs improvement: specifically, if the `true` return value should be removed. IIRC - but correct me if I'm wrong - these methods behave differently when called from an object context vs called statically.

Reply:
You're right, return type depends on whether the call is static :( Let's leave it alone for now then. Maybe add a comment?